lib/graphql/query.rb

```diff
@@ -28,7 +30,9 @@ def initialize(schema, query_string, context: nil, variables: {}, debug: false,
       @fragments = {}
       @operations = {}
       @provided_variables = variables
-      @document = GraphQL.parse(query_string)
+
+      @document = document
+      @document ||= GraphQL.parse(query_string)
```

Comment:
Would this also work:

```ruby
@document = document || GraphQL.parse(query_string) 
```

? 

I'd prefer to avoid reassigning the variable because a later reader may find the _first_ assignment but skip the second one!